resolwe/flow/managers/listener.py

```diff
@@ -48,58 +48,61 @@ def __init__(self, *args, **kwargs):
         super().__init__()
 
         # The Redis connection object.
-        self._redis = redis.StrictRedis(**kwargs.get('redis_params', {}))
+        self._redis = None
+        self._redis_params = kwargs.get('redis_params', {})
 
-        # An condition variable used to asynchronously signal when the
-        # listener should terminate.
-        self._terminated = Event()
+        # Running coordination.
+        self._should_stop = False
+        self._runner_coro = None
 
         # The verbosity level to pass around to Resolwe utilities
         # such as data_purge.
         self._verbosity = kwargs.get('verbosity', 1)
 
-    def clear_queue(self):
-        """Reset the executor queue channel to an empty state."""
-        self._redis.delete(state.MANAGER_EXECUTOR_CHANNELS.queue)
+    async def _make_connection(self):
+        """Construct a connection to Redis."""
+        return await aioredis.create_redis(
+            'redis://{}:{}'.format(
+                self._redis_params.get('host', 'localhost'),
+                self._redis_params.get('port', 56379)
```

Comment:
I will be much easier to debug if you don't specify default values. You can add a check on start to make sure these settings are set.